src/elements.ts

```diff
@@ -1354,51 +1354,105 @@ const SVG_TAGS = ['path', 'rect', 'circle', 'ellipse', 'polygon', 'polyline',
   'g', 'defs', 'marker', 'line', 'text', 'tspan', 'pattern', 'mask', 'svg',
   'foreignObject', 'image', 'use'];
 
+type SvgTag =
+  'path'|
+  'rect'|
+  'circle'|
+  'ellipse'|
+  'polygon'|
+  'polyline'|
+  'g'|
+  'defs'|
+  'marker'|
+  'line'|
+  'text'|
+  'tspan'|
+  'pattern'|
+  'mask'|
+  'svg'|
+  'foreignObject'|
+  'image'|
+  'use';
+
+type InputTag =
+  'input' |
+  'select' |
+  'textarea';
+
+type MediaTag = 'video' | 'audio';
+
+type DomName =
+  SvgTag |
+  InputTag |
+  MediaTag |
+  string;
+
+type DomQuery = Element | DomName;
+
+type CreateResult<T extends DomName> =
+  T extends 'svg' ? SVGParentView :
+  T extends SvgTag ? SVGView :
+  T extends 'canvas' ? CanvasView :
+  T extends 'form' ? FormView :
+  T extends InputTag ? InputView :
+  T extends MediaTag ? MediaView :
+  ElementView;
+
+type QueryResult<T extends DomQuery> =
+  T extends Element ? ElementView :
+  T extends string ? CreateResult<T> | undefined :
```

Comment:
At first glance this and line 1409 probably look redundant but without these extends conditions the linter/compiler will complain that `T` is not guaranteed to be compatible with `extends DomName`